infra/src/main/kotlin/fi/fta/geoviite/infra/publication/PublicationValidation.kt

```diff
@@ -409,126 +406,178 @@ private fun validateFrontJointTopology(
+            "${switchOrTrackLinkageKey(validatingTrack)}.switch-no-alignments-connected" to
+                localizationParams("switch" to switchName.toString())
+        },
         validateWithParams(
-            switchAlignmentsUnlinkedToNonduplicates.isEmpty() || trackBeingValidatedIsConnectedToFullAlignment,
+            (linkingQuality.none { it.hasSomethingLinked() } && !switchHasTopologicalConnections) ||
+                notLinked.isEmpty(),
             WARNING,
         ) {
-            val alignmentsString =
-                switchAlignmentsUnlinkedToNonduplicates.joinToString { alignment ->
-                    alignment.joinToString("-") { joint -> joint.intValue.toString() }
-                }
-            val key =
-                if (switchAlignmentsLinkedToOnlyDuplicates.isEmpty()) {
-                    "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-not-connected"
-                } else {
-                    "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-only-connected-to-duplicate"
-                }
+            "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-not-connected" to
+                localizationParams(
+                    "switch" to switchName.toString(),
+                    "alignments" to
+                        notLinked.joinToString(", ") { q ->
+                            q.switchAlignment.joints.joinToString("-") { it.intValue.toString() }
+                        },
+                )
+        },
+        validateWithParams(linkedOnlyToDuplicates.isEmpty(), WARNING) {
+            "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-only-connected-to-duplicate" to
+                localizationParams(
+                    "switch" to switchName.toString(),
+                    "locationTracks" to summarizeSwitchAlignmentLocationTrackLinks(linkedOnlyToDuplicates),
+                )
+        },
+        validateWithParams(linkedPartially.isEmpty(), WARNING) {
+            "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-partially-connected" to
+                localizationParams(
+                    "switch" to switchName.toString(),
+                    "locationTracks" to summarizeSwitchAlignmentLocationTrackLinks(linkedPartially),
+                )
+        },
+        validateWithParams(linkedMultiply.isEmpty(), WARNING) {
+            "${switchOrTrackLinkageKey(validatingTrack)}.switch-alignment-multiply-connected" to
+                localizationParams(
+                    "switch" to switchName.toString(),
+                    "locationTracks" to summarizeSwitchAlignmentLocationTrackLinks(linkedMultiply),
+                )
+        },
+    )
+}
 
-            key to localizationParams("locationTracks" to alignmentsString, "switch" to switchName.toString())
-        }
-    }
+private data class SwitchAlignmentLinkingQuality(
```

Comment:
Oletan että tästä on tarkoitus luoda olioita käytännössä tuolla alemmalla funktiolla, vai? Mietin kannattaisiko tämä jopa tehdä niin, että tämän rakentaja olisi privaatti ja tuo alempi rakennusfunktio olisi companion objectissa, ja luonnit hoidettaisiin sitä kautta. Silloin ei olisi vaaraa että joku lähtee luomaan näitä sen ulkopuolelta, koska oletettavasti niin ei olisi tarkoitus tehdä.